packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -57,180 +58,185 @@ import { Solarray } from "scripts/libraries/Solarray.sol";
 // Additionally, we intentionally use "Input" and "Output" terminology to clearly distinguish these
 // scripts from the existing ones that use the "Config" and "Artifacts" terminology.
 contract DeploySuperchainInput is CommonBase {
-    // The input struct contains all the input data required for the deployment.
-    // The fields must be in alphabetical order for vm.parseToml to work.
-    struct Input {
-        bool paused;
-        ProtocolVersion recommendedProtocolVersion;
-        ProtocolVersion requiredProtocolVersion;
-        Roles roles;
+    using stdToml for string;
+
+    // All inputs are set in storage individually. We put any roles first, followed by the remaining
+    // inputs. Inputs are internal and prefixed with an underscore, because we will expose a getter
+    // method that returns the input value. We use a getter method to allow us to make assertions on
+    // the input to ensure it's valid before returning it. We also intentionally do not use a struct
+    // to hold all inputs, because as features are developed the set of inputs will change, and
+    // modifying structs in Solidity is not very simple.
+
+    // Role inputs.
+    address internal _guardian;
+    address internal _protocolVersionsOwner;
+    address internal _proxyAdminOwner;
+
+    // Other inputs.
+    bool internal _paused;
+    ProtocolVersion internal _recommendedProtocolVersion;
+    ProtocolVersion internal _requiredProtocolVersion;
+
+    // These `set` methods let each input be set individually. The selector of an input's getter method
+    // is used to determine which field to set.
+    function set(bytes4 _sel, address _address) public {
+        require(_address != address(0), "DeploySuperchainInput: cannot set null address");
+        if (_sel == this.guardian.selector) _guardian = _address;
+        else if (_sel == this.protocolVersionsOwner.selector) _protocolVersionsOwner = _address;
+        else if (_sel == this.proxyAdminOwner.selector) _proxyAdminOwner = _address;
+        else revert("DeploySuperchainInput: unknown selector");
     }
 
-    struct Roles {
-        address guardian;
-        address protocolVersionsOwner;
-        address proxyAdminOwner;
+    function set(bytes4 _sel, bool _value) public {
+        if (_sel == this.paused.selector) _paused = _value;
+        else revert("DeploySuperchainInput: unknown selector");
     }
 
-    // This flag tells us if all inputs have been set. An `input()` getter method that returns all
-    // inputs reverts if this flag is false. This ensures the deploy script cannot proceed until all
-    // inputs are validated and set.
-    bool public inputSet = false;
```

Comment:
> Downside is it's more code to maintain and remember to update.

Yea this is my concern. I think we are already very generous about including safety checks elsewhere, which helps mitigate the need for this. Additionally, different consumers may need different inputs set, so consumers can just check the required inputs themselves if they need to verify upfront that everything is set. Examples:
- The switch from L2OO -> to FP, via inheritance: the L2OO setter and getter exist but now we don't need, so we must redefine the full `inputSet` method, instead of extending it
- The `deployAndInitializeSuperchainConfig` method: it only needs a subset of the inputs to be set, so it checks those only. Therefore when deploy functions are composed, all safety checks are already present anyway

I would say let's keep it out for simplicity and only add it back if we find a need for it.